src/publicKeyProvider/index.ts

```diff
@@ -0,0 +1,3 @@
+export * from './CachingKeyProvider';
+export * from './GotKeyProvider';
+export * from './KeyProvider';
```

Comment:
It looks somehow redundant to add 3 HTTP clients as dependencies of the authentications module. The more dependencies - the more hussle is expected in future when any of them get a vulnerability reported. I'd suggest removing all `KeyProvider` implementations altogether and leaving just a general-purpose caching wrapper. Implementations are easy and may be included as code snippets in README instead.